server/src/main/java/com/cloud/deploy/DeploymentPlanningManagerImpl.java

```diff
@@ -574,6 +580,80 @@ private boolean getDisplayStorageFromVmProfile(VirtualMachineProfile vmProfile)
         return vmProfile == null || vmProfile.getTemplate() == null || !vmProfile.getTemplate().isDeployAsIs();
     }
 
+        /**
+         * Adds disabled resources (Data centers, Pods, Clusters, and hosts) to exclude list (avoid) in case of disabled state.
+         */
+        public void avoidDisabledResources(VirtualMachineProfile vmProfile, DataCenter dc, ExcludeList avoids) {
+            if (vmProfile.getType().isUsedBySystem() && isRouterDeployableInDisabledResources()) {
+                return;
+            }
+
+            VMInstanceVO vm = _vmInstanceDao.findById(vmProfile.getId());
+            AccountVO owner = accountDao.findById(vm.getAccountId());
+            if (owner.getRoleId() == ADMIN_ACCOUNT_ROLE_ID && isAdminVmDeployableInDisabledResources()) {
```

Comment:
@DaanHoogland the problem raised by the tests is due to the fact that the role ID of a project is null; thus, these tests are failing.

Tests that failed:
/marvin/tests/smoke/test_affinity_groups_projects.py
/marvin/tests/smoke/test_kubernetes_clusters.py
/marvin/tests/smoke/test_projects.py
I am going to take a look at a fix for it, not sure if this will be on 4.15 then. Is there an ETA for 4.15?

Reply:
:) last month, @GabrielBrascher